packages/core/lib/datasources/Datasource.js

```diff
@@ -137,18 +137,25 @@ class Datasource extends EventEmitter {
     // Transform the received quads
     let destination = new BufferedIterator(), outputQuads, graph = this._graph;
     outputQuads = destination.map((quad) => {
+      let transformedSubject, transformedObject, transformedGraph;
       // Translate blank nodes in the result to blank node IRIs.
       if (quad.subject && quad.subject.termType === 'BlankNode' && !this._skolemizeBlacklist[quad.subject.value])
         quad.subject = this.dataFactory.namedNode(blankNodePrefix + quad.subject.value);
```

Comment:
This will still produce an error.

Reply:
Updated the subject and object assignments too.
Any other way this produces an error

Reply:
If it runs and the tests pass with the latest N3 version, then it should be ok I guess.